tests/functional/filter/cram/filter-empty-results-reporting.t

```diff
@@ -0,0 +1,39 @@
+Setup
+
+  $ source "$TESTDIR"/_setup.sh
+
+Filter using `--exclude-all` to easily get an empty result.
+
+Test the default behavior for empty results is an error.
+
+  $ ${AUGUR} filter \
+  >  --metadata "$TESTDIR/../data/metadata.tsv" \
+  >  --exclude-all \
+  >  --output-strains filtered_strains.txt > /dev/null
+  ERROR: All samples have been dropped! Check filter rules and metadata file format.
+  [2]
+  $ wc -l filtered_strains.txt
+  \s*0 .* (re)
+
+Repeat with the --empty-results-reporting=warn option.
+This whould output a warning message but no error.
+
+  $ ${AUGUR} filter \
+  >  --metadata "$TESTDIR/../data/metadata.tsv" \
+  >  --exclude-all \
+  >  --output-strains filtered_strains.txt \
+  >  --empty-results-reporting warn > /dev/null
+  WARNING: All samples have been dropped! Check filter rules and metadata file format.
+  $ wc -l filtered_strains.txt
+  \s*0 .* (re)
+
+Repeat with the --empty-results-reporting=skip option.
+This should not output any messages to stderr.
+
+  $ ${AUGUR} filter \
+  >  --metadata "$TESTDIR/../data/metadata.tsv" \
+  >  --exclude-all \
+  >  --output-strains filtered_strains.txt \
```

Comment:
Would you mind adding a test (or modifying this one) that outputs empty sequences and metadata along with empty strains? The first behavior I thought to check manually was whether the command also appropriately created empty versions of those outputs, too, but it would be nice if functional tests already covered that.

Reply:
Thanks for asking for this! Turns out we always write the metadata headers if the metadata output is requested! 

Does including the metadata headers even when results are empty make sense? Or should this behavior be changed to output a completely empty file?

Reply:
Sounds good! I'll just make sure the header matches the input header.

Reply:
Added the additional outputs to the Cram test in force push in 9df8b42fd7037e522f953fe232f5c3b1d8513632.